app/controllers/users_controller.rb

```diff
@@ -22,28 +22,13 @@ def show
 
   def create
     @user = User.new(user_params)
-    @invitation = Invitation.find_by(token_params)
-    @friend = @invitation.user if @invitation
-    if @user.valid?
-      charge = StripeWrapper::Charge.create(
-        stripe_params(
-          amount: 999,
-          description: "MyFlix sign up charge for #{@user.email}"
-        )
-      )
-      if charge.successful?
-        @user.save
-        mutual_friends if @friend
-        UserMailer.welcome(@user).deliver
-        sign_in @user
-        flash[:success] = "Signed up!"
-        redirect_to home_path
-      else
-        flash[:error] = charge.error_message
-        render :new
-      end
+    response = UserSignup.new(@user).sign_up(token_params, stripe_params)
+    if response.successful?
+      sign_in @user
+      flash[:success] = "Signed up!"
+      redirect_to home_path
     else
-      flash[:error] = "Sorry, can't complete sign up. Please correct any errors below."
+      flash[:error] = response.error_message
```

Comment:
It should be `flash.now` when you render. Otherwise the flash will pop on next GET request.